lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -673,15 +673,15 @@ def key(options = {})
         options[:key] || @api_key
       end
 
-      def headers(options = {})
+      def headers(method = :post, options = {})
         headers = {
           'Authorization' => 'Basic ' + Base64.strict_encode64(key(options).to_s + ':').strip,
           'User-Agent' => "Stripe/v1 ActiveMerchantBindings/#{ActiveMerchant::VERSION}",
           'Stripe-Version' => api_version(options),
           'X-Stripe-Client-User-Agent' => stripe_client_user_agent(options),
           'X-Stripe-Client-User-Metadata' => { ip: options[:ip] }.to_json
         }
-        headers['Idempotency-Key'] = options[:idempotency_key] if options[:idempotency_key]
+        headers['Idempotency-Key'] = options[:idempotency_key] if options[:idempotency_key] && method == :post
```

Comment:
This looks pretty straight forward.

I assume that stripe only accepts idempotency_key for post requests? It makes sense that they wouldn't have it on gets but there aren't any `put` requests or anything that we may run into later is there?

Reply:
Yes only for post request but I just say that also accept for delete. I'll updated to be unless `method == :get`